### PR TITLE
findbugs and a couple of warning fixes

### PR DESCRIPTION
**Pull request by adelnizamutdinov**
I was amazed that findbugs fails the build if `volatile` is missing on DI singletons

I'm also sorry for removing the `appContext` (for now?) but findbugs was yelling at me for that (and I respect that)

### CODE REVIEW

File: app/src/main/java/com/flatstack/android/DI.java [hunk 1]
```diff
@@ -44,14 +47,14 @@ public static OkHttpClient getHttpClient() {
 
     private static volatile Picasso picassoInstance = null;
 
-    public static Picasso getPicasso() {
+    public static @NonNull Picasso getPicasso(@NonNull Context context) {
         Picasso localInstance = picassoInstance;
```
Comment (IlyaEremin) [inline app/src/main/java/com/flatstack/android/DI.java]: What if Picasso will be called with ContextWrapper from `View`? Then `OkHttpClient` and `Picasso` singletons will be created with it. Any consequences?

Comment (adelnizamutdinov) [inline app/src/main/java/com/flatstack/android/DI.java]: nothing for getCacheDir, but 

- [ ] Picasso should probably be passed an app Context

Comment (adelnizamutdinov) [inline app/src/main/java/com/flatstack/android/DI.java]: oh wait, all good https://github.com/square/picasso/blob/master/picasso/src/main/java/com/squareup/picasso/Picasso.java#L724

**Comment by IlyaEremin** [comment]
:+1:

**Comment by adelnizamutdinov** [comment]
@IlyaEremin I just nuked the `DI` class, what do you think?

**Comment by IlyaEremin** [comment]
Leave it. I am going to add retrofit and related stuffs and I will need it

**Comment by adelnizamutdinov** [comment]
@IlyaEremin PTAL

**Comment by IlyaEremin** [comment]
Like it :+1: again

**Comment by adelnizamutdinov** [comment]
@IlyaEremin I don't have a merge permission

**Comment by IlyaEremin** [comment]
haha

**Pull request merged**